### DISCUSSION TITLE
mTLS working between Burp Suite -> API but not with Burp Suite -> mitmdump -> API

### DISCUSSION THREAD

**Discussion by maycon** [mitmproxy/mitmproxy#6585] [Q&A]
Hi,

I'm trying to use mitmdump to intercept and make some modifications on my requests, but for some reason the `--certs` and `--cert-passphrase` are not working for me.

I have a `mtls.pem` file containing the CERTIFICATE and the PRIVATE KEY. When configuring directly on the mitmdump I can make requests to my API. Even with a single `curl --cert mtls.pem` it works. But for some reason I cannot get it working with Burp Suite -> mitmdump -> API.

I am just doing:
```
$ mitmdump --ssl-insecure --listen-port 8282 --flow-detail 5 -s my-script.py --certs 'mtls.pem' --cert-passphrase xpto
```

The outgoing proxy is configured on Burp Suite.

When it do not works, I get a `401 Unauthorized` message, and when works I get a `200 OK`.

The Java SNI support in Burp is enabled, so I have no clues of what I am doing wrong.

**Comment by mhils** [comment]
Are you aiming to do mTLS with the API? In that case you probably want https://docs.mitmproxy.org/stable/concepts-options/#client_certs, and not the `certs` option. :)
